Libplanet/Assets/Currency.cs

```diff
@@ -671,9 +671,53 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
         /// <summary>
         /// Serializes the currency into a Bencodex value.
         /// </summary>
+        /// <param name="hash">Whether to serialize the currency for hash calculation.</param>
         /// <returns>The serialized Bencodex value.</returns>
         [Pure]
-        public IValue Serialize()
+        public IValue Serialize(bool hash = false) =>
```

Comment:
- Adding a parameter with a default value also will break binary compatibility (not source compatibility), so we need to mention it.
- default value of `hash` is `false` and it will be redirected `SerializeForState()`, but the previous behavior seems `SerializeForHash()`. is it intended change?

Reply:
There's a little bit of complexity here. `NineChronicles` was not using `Currency.Serialize` from `libplanet`. Instead, I was creating my own `CurrencyExtensions` to serialize/deserialize. Therefore, in order to read the Currency used by `NineChronicles` from `libplanet`, we had no choice but to bring the method of `NineChronicles` to `libplanet`.

This is a problem caused by `libplanet` not providing `Currency.Serialize()`, and rather, the existing `Currency.Serialize()` is a version that breaks backward compatibility of `NineChronicles`.

Reply:
So I chose to import the `Serialize` method of `NineChronicles`, and I wanted to delete the existing unused `Currency.Serialize()`, but this required a lot of modification at the code level, so I added a boolean parameter on `Serialize()` for much easier refactoring.

Reply:
Well... OK. if we can convince `Currency.Serialize()` hadn't been used outside of anywhere and we want to change its behavior like as `SerlizeForState()`, hiding `SerializeForHash()` and changing its behavior directly without new parameter introducing (i.e., https://github.com/planetarium/libplanet/pull/3201#discussion_r1227843653)  would be better than the current implementation.

Reply:
> Hiding SerializeForHash() and changing its behavior directly without a new parameter introducing 

I strongly agree with your opinion. I'll fix it.

Reply:
https://github.com/planetarium/libplanet/pull/3201/commits/2ad90d87437c3fd5d266da40052ef27a41fe3426 amended.